modules/preparing-vsphere-for-windows-containers.adoc

```diff
@@ -0,0 +1,157 @@
+// Module included in the following assemblies:
+//
+// * windows_containers/creating_windows_machinesets/creating-windows-machineset-vsphere.adoc
+
+[id="preparing-vsphere-for-windows-containers_{context}"]
+= Preparing your vSphere environment for Windows container workloads
+
+You must prepare your vSphere environment for Windows container workloads.
+
+== Creating the vSphere Windows VM golden image
+
+Create a vSphere Windows virtual machine (VM) golden image.
+
+.Prerequisites
+
+* You have installed a cluster on vSphere.
+
+.Procedure
+
+. Create the VM from an updated version of the Windows Server 1909 VM image that includes the following Microsoft link:https://support.microsoft.com/en-us/help/4565351/windows-10-update-kb4565351[patch].
+
+. Create the `C:\Users\Administrator.ssh\authorized_keys` file in the Windows VM containing the public key that corresponds to the private key that resides in the secret you created in the `openshift-windows-machine-config-operator` namespace. The private key of the secret was created when first installing the Windows Machine Config Operator (WMCO) to give {product-title} access to Windows VMs. The `authorized_keys` file is used to configure SSH in the Windows VM.
+
+. Configure SSH on the Windows VM by running the following Powershell script:
++
+----
+# Powershell script to configure SSH on vSphere Windows VMs
+Add-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0.1.0
+$firewallRuleName = "ContainerLogsPort"
+$containerLogsPort = "10250"
+New-NetFirewallRule -DisplayName $firewallRuleName -Direction Inbound -Action Allow -Protocol TCP -LocalPort $containerLogsPort -EdgeTraversalPolicy Allow
+Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force
+Install-Module -Force OpenSSHUtils
+Set-Service -Name ssh-agent -StartupType ‘Automatic’
+Set-Service -Name sshd -StartupType ‘Automatic’
+Start-Service ssh-agent
+Start-Service sshd
+$pubKeyConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace '#PubkeyAuthentication yes','PubkeyAuthentication yes'
+$pubKeyConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
+$passwordConf = (Get-Content -path C:\ProgramData\ssh\sshd_config) -replace '#PasswordAuthentication yes','PasswordAuthentication yes'
```

Comment:
Should we be allowing PasswordAuthentication? Seems like we should only allow key based auth

Reply:
@sebsoto this was a copy paste of https://github.com/openshift/windows-machine-config-operator/blob/master/docs/powershell.ps1, which was recommended in the [vSphere pre-requisites](https://github.com/openshift/windows-machine-config-operator/blob/master/docs/vsphere-prerequisites.md) doc.

I'm happy to update the script here to whatever should be promoted to customers. If it does need updating, please let me know of the new Powershell script to share.

Reply:
I agree with @codyhoag. If the script needs updating then it should be updated and tested in the WMCO repo first before reflecting it in the docs.

Reply:
FYI, I just did [a PR for the doc](https://github.com/openshift/windows-machine-config-operator/pull/273) since it requires some changes for the script to work.